### DISCUSSION TITLE
Allowing climate integrations to return None as HVAC_MODE

### DISCUSSION THREAD

**Discussion by janiversen** [home-assistant/architecture#616] [Entity Models] [answered]
A number of climate integrations do not handle HVAC_MODE, and therefore they do something like:
```
def hvac_mode(self):
        """Return hvac operation ie. heat, cool mode."""
        ...
        return None
```
Which is not allowed, and cause problems, having mypy produce errors correctly.
because in components/climate the definitions do not allow None:
```
@property
    def hvac_mode(self) -> str:
        """Return hvac operation ie. heat, cool mode.

        Need to be one of HVAC_MODE_*.
        """
        return self._attr_hvac_mode

 @property
    def state(self) -> str:
        """Return the current state."""
        return self.hvac_mode
```

It should be OK for an integration to tell "I do not know what the HVAC_MODE is", and there are 2 simple solutions:

a) Allow None:
```
@property
    def hvac_mode(self) -> str | None:
        """Return hvac operation ie. heat, cool mode.

        Need to be one of HVAC_MODE_*.
        """
        return self._attr_hvac_mode
```
Which would have ripple effects on "state" as well as the UI, and in general it is difficult to see where the ripples stop.

b) Add a new mode HVAC_MODE_UNKNOWN
that would not cause ripple effects in "state" etc, but still cause a ripple in the UI. 
In the UI we would need to handle this with e.g. a new icon.

Adding HVAC_MODE_UNKNOWN expands the current model, but so does allowing None (In reality), I believe adding HVAC_MODE_UNKNOWN is more readable and the changes needed are easy to see and implement..

We want to activate mypy, which have a problem with the current bad way of doing it (illegal None), and the current circumvention is to return "", which is equally bad.

This "discussion" originates from an old PR #50109 and a late review comment from @MartinHjelmare on #55077

**Comment by MartinHjelmare** [comment, marked as answer]
The application wide way of representing unknown state before the state is set in the state machine is `None`. So we should stick to that in my opinion.

**Comment by janiversen** [comment]
Seems accepting None is the way of least resistance.

How do we proceed? I submit a PR changing the HVAC_MODE definition to allow None ?

**Reply by MartinHjelmare** [reply]
It would be good with some more input from members especially regarding frontend compatibility.
